### DISCUSSION TITLE
refrain from giving focus to new windows

### DISCUSSION THREAD

**Discussion by gregrwm** [i3/i3#4554] [Q&A]
Upon reading the user's guide i'd hoped the following would refrain from giving focus to new windows:
```
focus_on_window_activation urgent
no_focus [window_type=dialog window_type=normal window_type=splash window_type=toolbar window_type=tooltip window_type=utility window_type=notification]
```
and per the first suggested answer i've also tried
```
focus_on_window_activation urgent
no_focus [window_type="dialog"]
no_focus [window_type="normal"]
no_focus [window_type="splash"]
no_focus [window_type="toolbar"]
no_focus [window_type="tooltip"]
no_focus [window_type="utility"]
no_focus [window_type="notification"]
```
but new windows still steal focus when they arrive.  eg i launch an app from an xterm, the app takes some time to appear, meanwhile i'm still interacting in the xterm, and when the app arrives it receives focus.

How may i configure i3 to refrain from giving focus to new windows?

is this perchance a bug for which the fix is available if i compile from source?  so far i'm running lubuntu 20.04 i3-wm:amd64 (4.17.1-1).
Complete current i3config:  [i3config.txt](https://github.com/i3/i3/files/7269747/i3config.txt)

**Comment by Airblader** [comment]
Command criteria are additive, so your current attempt won't ever match anything. If you want to match specifically on those criteria, you need to split it up into multiple `no_focus` directives.

**Comment by orestisfl** [comment]
Editing your post does not emit a notification so we can't track updates to this post.

For your issue, I suggest you open a new bug report with the latest i3 version and follow the instructions in the template.
